tmt/steps/__init__.py

```diff
@@ -828,12 +876,32 @@ def _patch_raw_datum(
 
             self.plan._applied_cli_invocations.append(invocation)
 
+        cli_invocations = self.__class__.cli_invocations
+
+        from tmt.steps import discover, execute, finish, prepare, provision, report
+
+        # Under some conditions, the step may need to have at least one phase
+        # defined.
+        if isinstance(self, (discover.Discover, execute.Execute, report.Report)):
+            if self.enabled:
+                raw_data = _ensure_default_phase()
+
+        elif isinstance(self, provision.Provision):
+            if self.enabled or cli_invocations and \
+                    any(invocation.options['how'] for invocation in cli_invocations):
+                raw_data = _ensure_default_phase()
+
+        elif isinstance(self, (prepare.Prepare, finish.Finish)):  # noqa: SIM102
```

Comment:
Perhaps a short explanation why `Prepare` and `Finish` steps do not need default phase unless `prepare --how` or `finish --how` are provided on the command line would be useful for future readers of the code?